File: src/lib-node/file-save.js
function file_save(x,y)
{
 check(is_str,x)
 check(is_str,y)
 
 if(is_file(x))
 {
  const s=file_read(x)
  
  if(same(s,y))
   return
 }
 
 const dir=path_dir(x)
 
 if(!is_dir(dir))
  dir_make(dir)
 
 file_write(x,y)
}

File: src/prj-core/lib-node/file/file-save.cs
fn file_save path:str data:str
 //same content

 if is_file path
  let s file_load path

  if same s data
   ret
 end

 //create directory tree

 let dir path_dir path

 if not is_dir dir
  dir_make dir

 //write

 file_write path data
end
